pkg/cmd/operator/gatherbase.go

```diff
@@ -59,10 +62,11 @@ func NewGatherBaseOptions(gathererName string, keepGoing bool) *GatherBaseOption
 func (o *GatherBaseOptions) AddFlags(flagset *pflag.FlagSet) {
 	o.ConfigFlags.AddFlags(flagset)
 
-	flagset.StringVarP(&o.DestDir, "dest-dir", "", o.DestDir, "Destination directory where to store the artifacts.")
-	flagset.Int64VarP(&o.LogsLimitBytes, "log-limit-bytes", "", o.LogsLimitBytes, "Maximum number of bytes collected for each log file, 0 means unlimited.")
-	flagset.BoolVarP(&o.CollectManagedFields, "managed-fields", "", o.CollectManagedFields, "Controls whether metadata.managedFields should be collected in the resource dumps.")
-	flagset.BoolVarP(&o.KeepGoing, "keep-going", "", o.KeepGoing, "Controls whether the collection should proceed to other resources over collection errors, accumulating errors.")
+	flagset.StringVar(&o.DestDir, "dest-dir", o.DestDir, "Destination directory where to store the artifacts.")
+	flagset.Int64Var(&o.LogsLimitBytes, "log-limit-bytes", o.LogsLimitBytes, "Maximum number of bytes collected for each log file, 0 means unlimited.")
+	flagset.BoolVar(&o.CollectManagedFields, "managed-fields", o.CollectManagedFields, "Controls whether metadata.managedFields should be collected in the resource dumps.")
+	flagset.BoolVar(&o.KeepGoing, "keep-going", o.KeepGoing, "Controls whether the collection should proceed to other resources over collection errors, accumulating errors.")
+	flagset.StringSliceVar(&o.ExcludeResources, "exclude-resources", []string{}, "List of resources to exclude from collection. Format: <kind>.<group>. Kind is expected to be in CamelCase (it's case sensitive).")
```

Comment:
nit: one thing I missed before, should we use o.ExcludeResources as the default (instead of inlined []string{})? And initialize it as part of the options constructor. Otherwise it's a bit all over the place with how the existing flags are initializied.

Reply:
How about doing it a bit differently and putting all the defaults in flag definitions? It seems counterintuitive to me to use the same variable for a default, and then for loading actual values from the CLI. It's also easier to grasp what the specific CLI flag is set to by default, having it this way.

Reply:
From correctness perspective, unless you initialise in the constructor,  the struct has uninitialised fields before the AddFlags method is called. So I personally prefer it this way.